File: algorithms/1491.average-salary-excluding-the-minimum-and-maximum-salary/solution.rb
# @param {Integer[]} salary
# @return {Float}
def average(salary)
  ### Use sort
  # salary.sort[1..-2].sum / (salary.size - 2).to_f

  min, max = salary[0], 0
  sum = 0
  salary.each do |s|
    min = s if s < min
    max = s if s > max
    sum += s
  end
  (sum - (min + max)) / (salary.size - 2).to_f
end

File: algorithms/1491.average-salary-excluding-the-minimum-and-maximum-salary/solution.rs
impl Solution {
    pub fn average(salary: Vec<i32>) -> f64 {
        let (mut min, mut max) = (salary[0], 0);
        let len = salary.len();
        let mut sum = 0;

        for s in salary {
            if s < min {
                min = s
            };
            if s > max {
                max = s
            };

            sum += s;
        }

        (sum - min - max) as f64 / (len - 2) as f64
    }
}
